flexmeasures/data/models/time_series.py

```diff
@@ -682,3 +683,83 @@ def parse_source_arg(
         else:
             parsed_sources.append(source)
     return parsed_sources
+
+
+class Annotation(db.Model):
```

Comment:
For code organization, it would be nice to split this large module up, or at least stop making it larger. This new code could go to `flexmeasures.data.models.annotations`.

(I also feel the `Sensor` class does semantically not belong under "time_series". We can split the models into business and data parts or so, if we don't like each class being in their own module.)

Reply:
For your convenience, I'd like to do this refactoring after you reviewed my other corrections.

I plan to move the 3 new classes related to annotations as you suggest, but to keep the `Sensor` class where it is for now. Two arguments:
- A semantic argument: the `Sensor` class describes essential attributes of the time series, and no business attributes (those belong to the `GenericAsset` class).
- A convenience argument: other packages depend on the `Sensor` class being where it is, so moving it would require us to think about a deprecation strategy (for its location). On that note, perhaps it makes sense to expose our most commonly used classes in a more central place, like: `from flexmeasures import Sensor`, similar to how we do it in https://github.com/SeitaBV/timely-beliefs/blob/main/timely_beliefs/__init__.py. We can then recommend importing it like that, and give ourselves some more flexibility in refactoring the location of the class (or even its name, which may be useful for renaming the `GenericAsset` class later).

Reply:
I agree it's easier for me this way to review, thanks.

Good idea to expose important models on a high level. I guess the other packages accessing them would usually be plugins right? What are the crucial models?

- Sensor
- Asset (-> GenericAsset)
- DataSource
- User
- Account

Reply:
Moved discussion to https://github.com/FlexMeasures/flexmeasures/issues/350.